packages/toc/src/toc_tree.tsx

```diff
@@ -65,10 +73,20 @@ class TOCTree extends React.Component<IProperties, IState> {
     // Map the heading objects onto a list of JSX elements...
     let i = 0;
     let list: JSX.Element[] = this.props.toc.map(el => {
+      const commands = new CommandRegistry();
+      const contextMenu = new Menu({ commands });
+
+      contextMenu.addItem({
+        args: { runCells: '' + NestedCodeCells },
+        command: CommandIDs.runCells
+      });
```

Comment:
To begin with something like:

```suggestion
      contextMenu.addItem({
        args: { toc:  this.props.toc, heading: el, code: [] },
        command: CommandIDs.runCells
      });
```

might be good enough. Again, this is not a tested code - just a draft to explain what I meant by defining command once and then passing arguments in `args`.